File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def getMinimumDifference(self, root: Optional[TreeNode]) -> int:
        if root is None:
            return None
        arr=[]
        stack=[]
        curr=root
        while(True):
            while curr is not None:
                stack.append(curr)
                curr=curr.left
            if stack:
                curr=stack.pop()
                arr.append(curr.val)
                curr=curr.right
            else:
                break
        ans=math.inf
        for i in range(1,len(arr)):
            ans=min(arr[i]-arr[i-1],ans)
        return ans
                
            
        

File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    void traverse(vector<int>&Arr,TreeNode* root)
    {
        if(root==nullptr)return;
        Arr.push_back(root->val);
        traverse(Arr,root->left);
        traverse(Arr,root->right);
    }
public:
    int getMinimumDifference(TreeNode* root) {
        vector<int>Arr;
        traverse(Arr,root);
        sort(Arr.begin(),Arr.end());
        int ans=INT_MAX;
        for(int i=0;i<Arr.size()-1;i++)
            ans=min(ans,Arr[i+1]-Arr[i]);
        return ans;
    }
};